### DISCUSSION TITLE
'ClerkProvider' cannot be used as a JSX component.

### DISCUSSION THREAD

**Discussion by eunoyaid** [clerk/javascript#4413] [Support request - DO NOT OPEN HERE]
always get error on my code layout
```
import "@/styles/globals.css";
import type { Metadata } from "next";
import { Providers } from "./providers";
import { fontSans } from "@/config/fonts";
import clsx from "clsx";
import { EdgeStoreProvider } from "@/lib/edgestrore";
import { ClerkProvider } from "@clerk/nextjs";
export const metadata: Metadata = {
  title: "Electrical Maintenance",
  description: "Electrical Maintenance",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <body className={clsx("font-sans antialiased", fontSans.className)}>
        <EdgeStoreProvider>
          <ClerkProvider>
            <Providers>{children}</Providers>
          </ClerkProvider>
        </EdgeStoreProvider>
      </body>
    </html>
  );
}
```

'ClerkProvider' cannot be used as a JSX component.
  Its return type 'Promise<React.JSX.Element>' is not a valid JSX element.
    Type 'Promise<React.JSX.Element>' is missing the following properties from type 'Element': type, props, keyts(2786)

**Comment by sucessdy** [comment]
Hi @eunoyaid,

I think u should Update the @clerk/nextjs Package with this command  `npm install @clerk/nextjs@latest`  and wrap ur clerk provider with async function maybe it's help or try to wrap ur component like this ... 😅
 
<img width="1250" alt="Screenshot 2024-11-15 at 11 02 20 PM" src="https://github.com/user-attachments/assets/59ef672f-f57b-4da8-9c70-34c99e6e5469">

**Comment by warrenrhodes** [comment]
Upgrade the version of  `@types/react` to ` 18.3.12 ` solve my problem.

**Reply by rajivk16** [reply]
thanks man! this fixed the issue for me! kudos!

**Reply by christensen143** [reply]
I had to downgrade to 18.3.12 to fix mine. I upgraded and it went to 18.3.16 and did not fix my issue.
